specs/v2-sketch.yaml

```diff
@@ -303,6 +348,22 @@ components:
         last_job_finish_time:
           type: string
           format: date-time
+    TriggerPipelineRequest:
+      type: object
+      additionalProperties: false
+      properties:
+        branch:
```

Comment:
I wonder if we should consider going with a unified "ref" format rather than making branch and tag top-level keys here??

Reply:
That might be nice, but I think we have behavioural differences depending on whether `foo` is a branch or a tag, so we may need to keep them separate for those reasons.

Reply:
Yes, at the moment you can specify `branch` and `revision` together, but `tag` must stand alone.
We can change it, but I'd like some time to think through the implications first.